irctest/server_tests/chmodes/secret.py

```diff
@@ -0,0 +1,50 @@
+from irctest import cases
+from irctest.numerics import RPL_LIST
+
+
+class SecretChannelTestCase(cases.BaseServerTestCase):
+    @cases.mark_specifications("RFC1459")
+    def testSecretChannelListCommand(self):
+        """
+        <https://datatracker.ietf.org/doc/html/rfc1459#section-4.2.6>
+
+        "Likewise, secret channels are not listed
+        at all unless the client is a member of the channel in question."
+        """
+
+        def get_listed_channels(replies):
+            channels = set()
+            for reply in replies:
+                # skip pseudo-channels (&SERVER, &NOTICES) listed by ngircd
+                # and ircu:
+                if reply.command == RPL_LIST and reply.params[1].startswith("#"):
+                    channels.add(reply.params[1])
+            return channels
```

Comment:
Could you share this function (or at least the filtering algo) with `server_tests/list.py`?